File: 은빈/자바/[PGS]비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i = 0; i < n; i++) {
            String str = "";
            int[] bin1 = getBinary(arr1[i], n);
            int[] bin2 = getBinary(arr2[i], n);
            
            for(int j = 0; j < n; j++) {
                if(bin1[j] == 1 || bin2[j] == 1) str += "#";
                else str += " ";
            }
            answer[i] = str;
        }
        
        return answer;
    }
    
    private int[] getBinary(int num, int size) {
        int[] arr = new int[size];
        String str = "";
        
        while(num != 0) {
            str += num % 2;
            num /= 2;
        }
        
        for(int i = 0; i < str.length(); i++) {
            arr[size - 1 - i] = str.charAt(i) - '0';
        }
        return arr;
    }
}

File: 유진/4월2주차/[PGS]비밀지도.js
function solution(n, arr1, arr2) {
  var answer = [];
  arr1 = arr1.map((item) => (item = item.toString(2).padStart(n, "0")));
  arr2 = arr2.map((item) => (item = item.toString(2).padStart(n, "0")));
  for (let a = 0; a < n; a++) {
    let result = "";
    for (let b = 0; b < n; b++) {
      if (arr1[a].charAt(b) == 1 || arr2[a].charAt(b) == 1) {
        result += "#";
      } else {
        result += " ";
      }
    }
    answer.push(result);
  }
  return answer;
}
